File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/linux/LinuxTextArea.java
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.linux;

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.TextArea;

public class LinuxTextArea implements TextArea {
    @Override
    public String getText() {
        return "linux text area";
    }
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/linux/LinuxTextArea.kt
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.linux

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.TextArea

class LinuxTextArea: TextArea {
    override fun getText(): String = "windows text area"
}